Kuadrant/kuadrant-controller
# ratelimitpolicy: new action types

## What
Added rate limit action types:
* [metadata](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-metadata)
* [remote_address](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-remoteaddress)
* [request_headers](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-requestheaders)

## Why

Currently only [generic_key](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-generickey) was available. Still other types are missing, for example, [header_value_match](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-headervaluematch)

New action types provide features for the configuration of the rate limit service

## Verification Steps

```
make local-setup
```

Create RateLimitPolicy and one virtual Service (toystory)

```
k apply -f examples/toystore
```

Annotate the VS to reference the RLP

```
k annotate virtualservice toystore kuadrant.io/ratelimitpolicy=toystore
```

Check gateway for injected rate limit configuration

```
istioctl proxy-config routes $(kubectl get pods -l istio=kuadrant-system -n kuadrant-system -o jsonpath='{.items[0].metadata.name}') -n kuadrant-system -o json | yq e -P

```
